src/core/manager.c

```diff
@@ -3736,6 +3818,72 @@ int manager_deserialize(Manager *m, FILE *f, FDSet *fds) {
         return manager_deserialize_units(m, f, fds);
 }
 
+int manager_pin_all_cgroup_bpf_programs(Manager *m) {
+        int r;
+        Unit *u;
+
+        assert(m);
+
+        HASHMAP_FOREACH(u, m->units) {
+                size_t i;
+                BPFProgram *p;
+                BPFProgram *builtin_progs[] = {u->bpf_device_control_installed,
+                                           u->ip_bpf_ingress,
+                                           u->ip_bpf_ingress_installed,
+                                           u->ip_bpf_egress,
+                                           u->ip_bpf_egress_installed};
+                Set *custom_progs[] = {u->ip_bpf_custom_ingress,
+                                       u->ip_bpf_custom_ingress_installed,
+                                       u->ip_bpf_custom_egress,
+                                       u->ip_bpf_custom_egress_installed};
+
+                for (i = 0; i < ELEMENTSOF(builtin_progs); i++) {
+                        p = builtin_progs[i];
+                        if (!p)
+                                continue;
+
+                        r = set_ensure_put(&m->bpf_limbo_progs, NULL, p);
+                        if (r < 0) {
+                                log_error_errno(r, "Cannot store BPF program for reload: %m");
+                                continue;
+                        }
+
+                        (void) bpf_program_ref(p);
```

Comment:
It is not necessary to voidify ref/unref function, as they do not return an error.